railties/lib/rails/generators/rails/app/templates/.devcontainer/devcontainer.json.tt

```diff
@@ -0,0 +1,34 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/ruby
+{
+  "name": "<%= app_name %>",
+  "dockerComposeFile": "docker-compose.yml",
+  "service": "rails-app",
+  "workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    "ghcr.io/devcontainers/features/github-cli:1": {}
+  },
+
+<%- if !devcontainer_variables.empty? -%>
+  "containerEnv": {
+    <%= devcontainer_variables.map { |key, value| "\"#{key}\": \"#{value}\"" }.join(",\n    ") %>
+  },
+<%- end -%>
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  // "features": {},
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  // "forwardPorts": [],
+
+  // Configure tool-specific properties.
+  // "customizations": {},
```

Comment:
If the target is to support Dev Containers in VS Code, then it could be valuable to add the Ruby LSP extension by default...

```json
"customizations": {
  "vscode": {
    "extensions": [
      "Shopify.ruby-lsp"
    ]
  }
}
```

This way, default debug configurations can also be added in `.vscode/launch.json` to debug the app and tests. I may have a working example of this setup in a private repo I could share, if there's interest.

Reply:
It is already in the ruby feature present in the base image.

Reply:
We decided to not add the launch.json yet because it isn't working in a way that is satisfactory to me. When I get my team to fix it, we will add all LSP features. Also see #50453